python/tank/platform/engine.py

```diff
@@ -504,7 +505,17 @@ def execute_in_main_thread(self, func, *args, **kwargs):
         else:
             # we don't have an invoker so just call the function:
             return func(*args, **kwargs)
-                
+
+    def renew_session(self):
+        """
+        Reauthenticate with the currently logged in user.
+        """
+        if self.has_ui:
+            # FIXME: Invoke from main thread.
```

Comment:
is this as simple as `self.execute_in_main_thread(interactive_authentication.ui_renew_session)` ?